File: app/src/main/java/test/project/MainMenu.java
package test.project;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainMenu extends AppCompatActivity {

    Button agregar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main_menu);

        agregar = (Button) findViewById(R.id.agregar);
        agregar.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent i = new Intent(MainMenu.this,addProduct.class);
                startActivity(i);
            }
        });
    }
}

File: app/src/main/java/test/project/MainMenu.kt
package test.project

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button

class MainMenu : AppCompatActivity() {

    internal var agregar: Button? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main_menu)

        agregar = findViewById(R.id.agregar) as Button
        agregar?.setOnClickListener {
            val i = Intent(this@MainMenu, addProduct::class.java)
            startActivity(i)
        }
    }
}
